### PR TITLE
fixes #824 - no reverse sort in apoc.coll.sortMaps

### LINKED ISSUE #824: No reverse sort in apoc.coll.sortMaps
The docs mention a possibility to get the sort order reversed by adding a `^` in front of the sorting field. 

The observed behavior is no sort at all (same as unrecognized field).

`WITH ['a','b','c','c','c','b','a','d'] AS l
RETURN apoc.coll.sortMaps(apoc.coll.frequencies(l),"count")`

gives `
[{"count":1,"item":"d"},{"count":2,"item":"a"},{"count":2,"item":"b"},{"count":3,"item":"c"}]                                               `

while `WITH ['a','b','c','c','c','b','a','d'] AS l
RETURN apoc.coll.sortMaps(apoc.coll.frequencies(l),"count")`
gives 
`[{"count":2,"item":"a"},{"count":2,"item":"b"},{"count":3,"item":"c"},{"count":1,"item":"d"}] `

### PR DESCRIPTION
**Pull request by AngeloBusato**

### CODE REVIEW

**Comment by jexp** [comment]
It should default to descending and  `^` means ascending.

**Comment by AngeloBusato** [comment]
@jexp PR updated

File: src/main/java/apoc/coll/Coll.java [hunk 1]
```diff
@@ -440,19 +442,26 @@ public long sumLongs(@Name("numbers") List<Number> list) {
     @Description("apoc.coll.sortNodes([nodes], 'name') sort nodes by property")
     public List<Node> sortNodes(@Name("coll") List<Node> coll, @Name("prop") String prop) {
         List<Node> sorted = new ArrayList<>(coll);
-        Collections.sort(sorted,
-                (x, y) -> compare(x.getProperty(prop, null), y.getProperty(prop, null)));
+        Collections.sort(sorted, (x, y) -> reverseOrder(prop) * compare(x.getProperty(cleanProperty(prop), null), y.getProperty(cleanProperty(prop), null)));
```
Comment (jexp) [inline src/main/java/apoc/coll/Coll.java]: reverse order and cleanProperty should happen before the sort loop.

File: src/main/java/apoc/coll/Coll.java [hunk 2]
```diff
@@ -440,19 +442,26 @@ public long sumLongs(@Name("numbers") List<Number> list) {
...
         return sorted;
     }
 
     @UserFunction
     @Description("apoc.coll.sortMaps([maps], 'name') - sort maps by property")
     public List<Map<String,Object>> sortMaps(@Name("coll") List<Map<String,Object>> coll, @Name("prop") String prop) {
         List<Map<String,Object>> sorted = new ArrayList<>(coll);
-        sorted.sort((x, y) -> compare(x.get(prop), y.get(prop)));
+        Collections.sort(sorted, (x, y) -> reverseOrder(prop) * compare(x.get(cleanProperty(prop)), y.get(cleanProperty(prop))));
```
Comment (jexp) [inline src/main/java/apoc/coll/Coll.java]: same here

**Comment by AngeloBusato** [comment]
@jexp PR Updated

**Comment by jexp** [comment]
Thanks a lot!

**Pull request merged**